lib/Controller/Schedule.php

```diff
@@ -344,7 +346,14 @@ public function eventData(Request $request, Response $response)
                 && $this->isEventEditable($row);
 
             // Event Title
-            if ($row->campaignId == 0) {
+            if ($this->isSyncEvent($row->eventTypeId)) {
+                $this->getDispatcher()->dispatch(new ScheduleLoadEvent($row), ScheduleLoadEvent::$NAME);
```

Comment:
This potentially generates a lot of events and I wonder if we need to do this in this way. For example, can we instead add to the originating query (perhaps optionally via a parameter), and just name the sync group.